services/AttendanceService.ts

```diff
@@ -71,6 +76,36 @@ export default class AttendanceService {
     });
   }
 
+  public async attendViaExpressCheckin(attendanceCode: string, email: string): Promise<PublicExpressCheckin> {
+    return this.transactions.readWrite(async (txn) => {
+      const expressCheckinRepository = Repositories.expressCheckin(txn);
+      const userRepository = Repositories.user(txn);
+
+      const hasUserUsedExpressCheckin = await expressCheckinRepository.hasUserUsedExpressCheckin(email);
+      if (hasUserUsedExpressCheckin) {
+        throw new UserError(
+          'You have already done an express check-in before. '
```

Comment:
I mentioned this in the issue but minor tweak to this error message: If they used Express Checkin for **this specific event,** error to let them know they're all good with checking in already, otherwise send this message